storage-units-model/src/main/java/wtf/metio/storageunits/model/StorageUnit.java

```diff
@@ -563,6 +563,86 @@ public final long longValue() {
         return bytes.longValue();
     }
 
+    /**
+     * Returns the value in this unit.
+     * @param roundingMode - the rounding mode for the division that divides the
+     *                     underlying byte value by number of bytes in a single measure of this unit.
+     *                     This parameter has no effect unless the unit has exuberant bytes per unit value.
+     * @return value in this unit.
+     */
+    @CheckReturnValue
+    public final BigDecimal unitValue(RoundingMode roundingMode) {
+        var divisor = new BigDecimal(getNumberOfBytesPerUnit());
+        var result = new BigDecimal(bytes).divide(divisor, conversionScale(), roundingMode).stripTrailingZeros();
+        if (result.scale() < 0) { result = result.setScale(0); }
+        return result;
+    }
+
+    protected abstract int conversionScale();
+
+    protected static int computeFiniteConversionScale(BigInteger bpu) {
+
+        // $TODO Use of static is ugly.
+        // But there is no lazy initialization in this library.
+        // One can stuff this into a final's field assignment, but
+        // assigning final fields from methods is evil. Moving to Kotlin,
+        // would be "better" alternatives.
+
+        // the default implementation is good for units provided by this library,
+        // because it either x1000 or x1024, where the division always yields a
+        // regular fraction. If any unit that has an exuberant BPU is introduced,
+        // they should override this method.
```

Comment:
since you are complaining here that we have no lazy init, how about we remove this entire method and hard-code the result of it in each `conversionScale` method of each unit? Since the result only depends on the number of bytes in each unit, this won't ever change for a unit anyway, so no need to compute it at runtime?

Reply:
I thought about it, and kinda decided to go with "let the computer do the calculations", to avoid it done menially, plus this method is still needed for tests to ensure that the scale is correct. If there was a lazy init, there would be a better way of handling it, avoiding a static variable in each implementation.

If you insist this was rather pre-computed, I can make it so.

Reply:
I was just doing lots of rust lately and kinda like this idea of running part of your code at compile time :sweat_smile: so I'm fine with the way it is and just wanted to provide an alternative solution

Reply:
yep all good here! I'll keep the comment around as-is, maybe a future version of Java will offer something new in this field :shrug: